[GEP-31] Add usage doc for in-place update (#12005)
gardener/gardener: docs/usage/shoot-operations/shoot_credentials_rotation.md

Once this list is empty, the `phase` transitions to `Prepared`.
Now you can just complete the rotation as usual (see above).

#### Worker Node with ManualInPlaceUpdate Update Strategy

> In case of manual in-place update, shoot CA rotation phase will be at `Preparing` until all the worker pools are successfully in-place updated and there are no pending worker pools with strategy ManualInPlaceUpdate.

Reviewer: ```suggestion
If the `Shoot` contains workers with strategy `ManualInPlaceUpdate`, the CA rotation phase will be at `Preparing` until all those worker pools are selected for update (See docs/usage/shoot-operations/shoot_updates.md#ManualInPlaceUpdate) and the update is completed. 
```

Reviewer: Add a section in the `ServiceAccount` signing key rotation below as well.

> You can check which worker pools still need to be updated by reading `.status.inPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate`.

Reviewer 2: ```suggestion
You can check which worker pools still need to be updated by reading `.status.inPlaceUpdates.pendingWorkerUpdates.manualInPlaceUpdate`.
```

> Once this list is empty, the `phase` transitions to `Prepared`.After this rotation will be completed as usual (see above).

Reviewer 3: ```suggestion
Once this list is empty, the `phase` transitions to `Prepared`. After this rotation will be completed as usual (see above).
```
```suggestion
Once this list is empty, the `phase` transitions to `Prepared`. After this rotation will be completed as usual (see above).
```

> Once this list is empty, the `phase` transitions to `Prepared`. After this rotation will be completed as usual (see above).

Reviewer 2: ```suggestion
Once this list is empty, the `phase` transitions to `Prepared`.
After this, rotation will be completed as usual (see above).
```

generally: one line per sentence

Please read [this section](#triggering-worker-node-rollout-individually) to get more information.
It works the same way for the `ServiceAccount` token signing key (using `rotate-serviceaccount-key-start-without-workers-rollout`).

#### Worker Node with ManualInPlaceUpdate Update Strategy

Similar to the rotation of the certificate authorities, in case of manual in-place update, `ServiceAccount` token signing key rotation phase will be at `Preparing` until all the worker pools are successfully in-place updated and there are no pending worker pools with strategy ManualInPlaceUpdate.

> Please read [this section](#worker-node-with-manualinplaceupdate-update-strategy) to get more information.

Reviewer 3: ```suggestion
Please read [this section](#worker-node-with-manualinplaceupdate-update-strategy) for more information.
```
